src/inmanta/server/services/notificationservice.py

```diff
@@ -63,14 +63,22 @@ async def start(self) -> None:
     async def _cleanup(self) -> None:
         await data.Notification.clean_up_notifications()
 
-    async def compile_done(self, compile: data.Compile) -> None:
+    async def compile_done(self, compile: data.Compile, notify_failed_compile: bool = False) -> None:
         if not compile.success:
             compile_report = await data.Compile.get_report(compile_id=compile.id)
             reports = compile_report["reports"] if compile_report else []
             failed_pull_stage = next(
                 (report for report in reports if report["name"] == "Pulling updates" and report["returncode"] != 0), None
             )
-            if failed_pull_stage:
+            if notify_failed_compile:
+                await self.notify(
+                    compile.environment,
+                    title="Compile request failed",
```

Comment:
```suggestion
                    title="Compilation failed",
```
Stay consistent with the existing behavior where we can?